lib/src/ui.dart

```diff
@@ -136,6 +136,20 @@ class NetworkLoggerScreen extends StatelessWidget {
     );
   }
 
+  final TextEditingController searchController =
+      TextEditingController(text: null);
+      
+  /// filte events with search keyword
+  List<NetworkEvent> getEvents() {
+    final events = eventList.events.where((element) {
+      if (searchController.text.length > 0) {
+        return (element.request?.uri.toLowerCase().contains(searchController.text.toLowerCase()) ?? false);
+      }
+      return true;
```

Comment:
I would write this like that:

```dart
  List<NetworkEvent> getEvents() {
    if (searchController.text.isEmpty) return eventList.events;
    final query = searchController.text.toLowerCase();
    return eventList.events.where((it) => (it.request?.uri.toLowerCase().contains(query) ?? false));
   }
```

- Reduces enumeration if search query is empty
- Reduces calls to `searchController.text.toLower()`

Reply:
Thanks for the suggestion, the code has been modified